app/composer.json

```diff
@@ -69,7 +69,7 @@
     "knplabs/gaufrette": "~0.9.0",
     "aws/aws-sdk-php": "~3.0",
     "friendsofsymfony/rest-bundle": "^3.0.2",
-    "friendsofsymfony/oauth-server-bundle": "dev-master",
+    "friendsofsymfony/oauth-server-bundle": "dev-doctrine-fix",
```

Comment:
https://github.com/FriendsOfSymfony/FOSOAuthServerBundle/branches I do not see this branch being available here - should it maybe be https://github.com/mautic/FOSOAuthServerBundle/branches that we are referring to?

If so, should we publish our fork (if it is still required) to Packagist to make it composer-friendly, or have it as a Git repo included?

Reply:
There is a Composer hack:

https://getcomposer.org/doc/05-repositories.md#loading-a-package-from-a-vcs-repository

That is saying to look into https://github.com/mautic/FOSOAuthServerBundle instead of  https://github.com/FriendsOfSymfony/FOSOAuthServerBundle. That is what I missed in https://github.com/mautic/mautic/pull/11353 and thought that the branch was deleted in the original repository because I couldn't find it.

See https://github.com/mautic/mautic/blob/1c38f1470158db5b51e7dbca51ef0f8370a9ed12/app/composer.json#L115

But it exists in https://github.com/mautic/FOSOAuthServerBundle/tree/doctrine-fix. The `dev-` prefix is there to say Composer that it is a branch, not a version/tag.

I'm not sure how it works with publishing clones on Packagist. If it is OK or if we should ask the original authors for permission? But it would make this more straight forward to wrap our heads around it and it would also speed up `composer install` because loading dependencies from repos is slow.